lib/class-wp-theme-json-gutenberg.php

```diff
@@ -1848,6 +1848,21 @@ protected static function compute_style_properties( $styles, $settings = array()
 
 		$root_variable_duplicates = array();
 
+		// Add a solid style fallback if no style defined.
+		if ( isset( $styles[ 'border'] ) ) {
+			if ( 
+				( isset( $styles[ 'border']['width'] ) || isset( $styles[ 'border']['color'] ) ) && ! isset( $styles[ 'border']['style'] ) ) {
+				$styles[ 'border']['style'] = 'solid';
+			}
+			foreach ( [ 'top', 'left', 'right', 'bottom' ] as $edge ) {
+				if (
+					isset( $styles[ 'border'][ $edge ]) && 
+					( isset( $styles[ 'border'][ $edge ]['width'] ) || isset( $styles[ 'border'][ $edge ]['color'] ) ) && ! isset( $styles[ 'border'][ $edge ]['style'] ) ) {
+					$styles[ 'border'][ $edge ]['style'] = 'solid';
+				}
+			}
+		}
```

Comment:
I think this "declarative approach" of one style object property generates one CSS rule is showing its limit in the style generation. There's so much edge cases in this file and it keeps growing. The other point here is that block supports do the exact same thing, so the code for style generation is duplicated.

The frontend side is a bit better and I think this highlights that we should move to a similar approach in the backend. (Style generation engine applied for both block supports and theme.json style generation). Not sure how that work is going.